feat(popupconnected): Align to right

Martinesp opened the pull request:
Added input to align the popup to right side, left side alignment is still default.
+ Updated example with no scroll and margin for arrow.

Martinesp committed:
feat(popupconnected): added input option to align popup to right direction

Updated example, and updated margin for arrow

JGH153 commented on projects/hal-components/src/lib/components/popup-connected/popup-connected.component.html:
```diff
@@ -9,7 +9,9 @@
-    <div class="arrow-above" *ngIf="relativePositionY === 'below'"></div>
+    <div 
+      [ngClass]="!alignedRight ? 'arrow-above' : 'arrow-above-right'"
+      *ngIf="relativePositionY === 'below'"></div>
```
bør ha ngIf før ngClass, samme under

JGH153 commented on src/app/doc-pages/popup-connected-doc/popup-connected-doc.component.html:
```diff
@@ -1,72 +1,99 @@
+  </p>
+  <p class="doc-info">
+    PopupConnectedComponent is a popup which is connected to an origin element.
+    The default position of the popup is above the origin it is connected to.
```
Er ikke default below?

JGH153 commented on projects/hal-components/src/lib/components/popup-connected/popup-connected.component.ts:
```diff
@@ -13,45 +11,66 @@ export type popupPosition = 'above' | 'below';
   @Input() isOpen = false;
   @Input() relativePositionY: popupPosition = 'below';
   @Input() origin: CdkOverlayOrigin;
+  @Input() alignedRight = false;
```
Tror vi bør nytte til alignRight 

Og enten må vi endre relativePositionY til allignTop, eller endre alignRight til relativePositionX. Stemmer for allignTop og alignRight. Det blir en liten breaking, men da får vi samme interface for begge

Martinesp replied:
Enig, har endret relativePositionY til alignTop nå. Endret også alignTop til å ta imot en boolean istedenfor så begge 'posisjons' inputene er er like i navn og inputtype

Martinesp committed:
improvement(popupconnected): changed relativeY string value to alignTop boolean value

Small doc update and fixes according to PR change requests

BREAKING CHANGE: Input relativePositionY is no longer valid input. To put the popup above the origin
point pass the alignRight input with the boolean value of true.

Martinesp committed:
Test(PopupConnected): updated test and package version

Martinesp committed:
test(popupconnected): fixed tests, and added test for alignRight conditions

JGH153 commented on projects/hal-components/package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@hafslundnett/hal-components",
-  "version": "2.2.1",
+  "version": "3.0.0",
```
Husk å skriv en breaking change som i HDD CHANGELOG.MD

Martinesp committed:
docs: added change log to project

JGH153 committed:
Merge branch 'trunk' into popupConnected-aligned-right

JGH153 committed:
Merge branch 'trunk' into popupConnected-aligned-right
